### PR TITLE
[v1.8] env_project_file correctly handle directories

### PR DESCRIPTION
**Pull request by simonbyrne**
On 1.7 and 1.8, if you provide a directory to `env_project_file`, it will return `true` even if there is a project file in the directory.

Note that this PR is against the release-1.8 branch: it was fixed on master by #44318, so we could also backport that one.

### CODE REVIEW

**Comment by KristofferC** [comment]
I changed the branch to be against the backport branch (so it will go through PkgEval) and apparently that messed things up. Perhaps it needs to be rebased... Sorry.

**Review by KristofferC — APPROVED** [review]
I guess a test would be good if it isn't too much effort.

**Comment by KristofferC** [comment]
Seems like testing fails on e.g. tester_win64.

**Comment by vtjnash** [comment]
It fails on all systems. Seems like the test is just written incorrectly?

File: test/loading.jl [hunk 1]
```diff
@@ -156,6 +156,16 @@ end
...
             """)
+
+            @test Base.env_project_file(joinpath(dir, "Root")).uuid == root_uuid
+            @test Base.env_project_file(joinpath(project_file, "Root")).uuid == root_uuid
+
+            @test Base.env_project_file(joinpath(dir, "This")).uuid == this_uuid
+            @test Base.env_project_file(joinpath(project_file, "This")).uuid == this_uuid
+
+            @test Base.env_project_file(joinpath(dir, "That")).uuid == nothing
+            @test Base.env_project_file(joinpath(project_file, "That")).uuid == nothing
```
Comment (vtjnash) [inline test/loading.jl:167]:
```suggestion
            @test Base.env_project_file(joinpath(dir)) == project_file
            @test Base.env_project_file(joinpath(project_file)) == project_file

            @test Base.env_project_file(joinpath(dir, "This")) === false
            @test Base.env_project_file(joinpath(project_file, "This")) === false

            mkdir(joinpath(dir, "Inner"))
            @test Base.env_project_file(joinpath(dir, "Inner")) === true
```

**Pull request closed without merging**